libs/ui-lib/lib/ocm/components/clusterConfiguration/platformIntegration/ExternalPlatformDropdown.tsx

```diff
@@ -43,8 +44,29 @@ export type ExternalPlatformInfo = {
   disabledReason?: string;
 };
 
+const getDisabledReasonForExternalPlatform = (
+  featureId: FeatureId,
+  isSNO: boolean,
+  isNutanixOrVsphere: boolean,
+  label: string,
+  newFeatureSupportLevelContext: NewFeatureSupportLevelData,
+  featureSupportLevelData?: NewFeatureSupportLevelMap | null,
+  cpuArchitecture?: string,
+): string | undefined => {
+  if (!isSNO) {
+    return newFeatureSupportLevelContext.getFeatureDisabledReason(
+      featureId,
+      featureSupportLevelData ?? undefined,
+      cpuArchitecture,
+    );
+  } else if (isSNO && isNutanixOrVsphere) {
```

Comment:
If you check the `!isSNO` condition in the _if_ branch, there is no need to check the opposite in the _else_ branch.

```suggestion
  } else if (isNutanixOrVsphere) {
```